pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/assembly/AssemblyVersionUpload.java

```diff
@@ -62,10 +62,7 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
             String locale = ServletUtils.paramValue(request, "locale", GlobalConfig.DEFAULT_MODULE_LOCALE.toString());
 //            String contentType = ServletUtils.paramValue(request, "type", "assembly");
 
-            String encoding = request.getCharacterEncoding();
-            if (encoding == null) {
-                encoding = StandardCharsets.UTF_8.name();
-            }
+            String encoding = StandardCharsets.UTF_8.name();
```

Comment:
@rednitish I'm worried that we're breaking support for non-UTF-8 encodings by doing this. Can you share your thought process behind this change?

Reply:
@benradey if you check encoding, after running uploader script, you will find, it's always ISO_8859_1, There is no use of this conditional block with current code, as it will never execute.

secondly, if you check com.redhat.pantheon.servlet.ServletUtils#handleParamAsStream(org.apache.sling.api.SlingHttpServletRequest, java.lang.String, java.util.function.Function<java.io.InputStream,R>) , line no 184, you can see, we are parsing content in UTF_8 while storing it in nodes.

Accordingly, i understand, we can go ahead by removing this block, to fix the encoding issue.

Reply:
I think you have struck at the heart of the bug with the lines you reference, but I think that the mechanisms at play are working a bit differently than you describe.

Take a look at how the code used to look in this class:
https://github.com/redhataccess/pantheon/blame/e672d87df1f7f221367aa06fa45daca1f23289ca/pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/assembly/AssemblyVersionUpload.java

The relevant line is 53 - it's very clear: "decode the string of bytes that I got from the request using the specified encoding, and give me a string re-encoded in UTF-8 because that's what I want to work with."

As you point out, the specified encoding was always ISO_8859_1 (but still, the 'if' block left the door open for other possible encodings). This code worked, but it was inefficient.

What you're proposing is to assume that the string of bytes coming from the request is encoded in UTF-8 from the start, which I believe is just not true. It "appears" to work but ignoring the specified encoding is unsafe. We really need to rely on the encoding that the request specifies being accurate.

But, you point at line 184 in ServletUtils as treating the input as UTF-8 by default. Honestly I can't really tell what's going on in the code. Method A provide Function B to Method C, which provides an InputStreamReader to Function B and executes it by wrapping a stream with a reader with a stream, but then Function B takes that stream and wraps it in another reader wrapped by another stream before finally wrapping it with another reader, *and then finally* applying that to the function that the developer specified, and the net result is...? Your guess is as good as mine. But you rightly point out that somewhere along the way, UTF-8 is brought in as a hardcoded default. I think THAT's what's mucking with our encoding and THAT's what needs to change.

@carlosmunoz since you wrote the code in question, can you help untangle what's actually going on in the ServletUtils class and whether this is likely where our encoding bug actually lies?

Also tagging @xdavidson and @aprajshekhar in case they can help sort this out.

Reply:
So, there are 2 methods in that ServletUtil class: one takes a charset parameter, the other one doesn't (uses UTF-8 by default). This is the character set to use when reading the stream.
Does that make sense?

Reply:
@rednitish Please remove the `encoding` variable completely and all subsequent references to it.